apps/datafeeder/src/app/presentation/pages/analysis-progress-page/analysis-progress.page.ts

```diff
@@ -31,19 +31,17 @@ export class AnalysisProgressPageComponent implements OnInit, OnDestroy {
   ngOnInit(): void {
     this.subscription = new Subscription()
     this.statusFetch$ = this.activatedRoute.params.pipe(
-      switchMap(({ id }) => {
-        return interval(250).pipe(
-          switchMap(() => {
-            return this.fileUploadApiService.findUploadJob(id)
-          }),
+      switchMap(({ id }) =>
+        interval(250).pipe(
+          switchMap(() => this.fileUploadApiService.findUploadJob(id)),
           tap((job: UploadJobStatusApiModel) => (this.progress = job.progress)),
-          takeWhile(
+          filter(
             (job: UploadJobStatusApiModel) =>
-              [PENDING, ANALYZING].includes(job.status),
-            true
-          )
+              ![PENDING, ANALYZING].includes(job.status)
+          ),
+          take(1)
```

Comment:
You don't want to get to onJobFinished until it's done or error. It used to work (by accident) because analysis was quick enough to be completed on first tick.